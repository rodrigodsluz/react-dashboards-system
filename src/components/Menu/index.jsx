import React from 'react';
import { Link } from 'react-router-dom';

import { useSelector, shallowEqual } from 'react-redux';

import Paper from '@material-ui/core/Paper';
import Divider from '@material-ui/core/Divider';
import MenuList from '@material-ui/core/MenuList';
import MenuItem from '@material-ui/core/MenuItem';
import Typography from '@material-ui/core/Typography';
import ListItemIcon from '@material-ui/core/ListItemIcon';

import GroupIcon from '@material-ui/icons/Group';
import EventNoteIcon from '@material-ui/icons/EventNote';
import AssessmentIcon from '@material-ui/icons/Assessment';
import AccountCircleIcon from '@material-ui/icons/AccountCircle';
import DeveloperModeIcon from '@material-ui/icons/DeveloperMode';
import AssignmentTurnedInIcon from '@material-ui/icons/AssignmentTurnedIn';

import useStyles from './styles';

const Menu = () => {
  const classes = useStyles();

  const user = useSelector((state) => state.User.user, shallowEqual);

  return (
    <Paper className={classes.root} elevation={3} variant="outlined">
      {user && (
        <MenuList className={classes.menuList}>
          <Link to="/dashboard" className={classes.link}>
            <MenuItem className={classes.menuItem}>
              <ListItemIcon>
                <AssessmentIcon className={classes.icon} />
              </ListItemIcon>
              <Typography variant="inherit" className={classes.linkLabel}>
                Dashboard
              </Typography>
            </MenuItem>
          </Link>
          <Divider />
          <Link to="/reports" className={classes.link}>
            <MenuItem className={classes.menuItem}>
              <ListItemIcon>
                <EventNoteIcon className={classes.icon} />
              </ListItemIcon>

              <Typography variant="inherit" className={classes.linkLabel}>
                Processos
              </Typography>
            </MenuItem>
          </Link>
          <Divider />
          {user.admin && (
            <Link to="/users" className={classes.link}>
              <MenuItem className={classes.menuItem}>
                <ListItemIcon>
                  <AccountCircleIcon className={classes.icon} />
                </ListItemIcon>
                <Typography variant="inherit" className={classes.linkLabel}>
                  UsuÃ¡rios
                </Typography>
              </MenuItem>
            </Link>
          )}
          {user.admin && <Divider />}
          {user.admin && (
            <Link to="/groups" className={classes.link}>
              <MenuItem className={classes.menuItem}>
                <ListItemIcon>
                  <GroupIcon className={classes.icon} />
                </ListItemIcon>
                <Typography variant="inherit" className={classes.linkLabel}>
                  Grupos
                </Typography>
              </MenuItem>
            </Link>
          )}
          {user.admin && <Divider />}
          {user.admin && (
            <Link to="/status" className={classes.link}>
              <MenuItem className={classes.menuItem}>
                <ListItemIcon>
                  <AssignmentTurnedInIcon className={classes.icon} />
                </ListItemIcon>
                <Typography variant="inherit" className={classes.linkLabel}>
                  Status
                </Typography>
              </MenuItem>
            </Link>
          )}
          {user.admin && <Divider />}

          {user.admin && (
            <Link to="/template" className={classes.link}>
              <MenuItem className={classes.menuItem}>
                <ListItemIcon>
                  <DeveloperModeIcon className={classes.icon} />
                </ListItemIcon>
                <Typography variant="inherit" className={classes.linkLabel}>
                  Template
                </Typography>
              </MenuItem>
            </Link>
          )}
          {user.admin && <Divider />}
        </MenuList>
      )}
    </Paper>
  );
};

export default Menu;
