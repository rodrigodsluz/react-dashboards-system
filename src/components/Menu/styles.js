import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  root: {
    width: 250,
    minWidth: 200,
    height: '100vh',
    borderRadius: 0,
  },
  menuList: {
    outline: 'none',
  },
  menuItem: {
    padding: '10px 20px',
  },
  link: {
    textDecoration: 'none',
    color: theme.palette.text.secondary,
  },
  linkLabel: {
    // fontWeight: '600',
  },
  icon: {
    color: theme.palette.text.secondary,
  },
}));

export default useStyles;
