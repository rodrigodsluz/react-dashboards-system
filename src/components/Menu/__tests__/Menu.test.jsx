import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

import Menu from "../index";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Menu />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
