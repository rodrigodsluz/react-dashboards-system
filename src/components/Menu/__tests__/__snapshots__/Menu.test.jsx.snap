// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
>
  <ul
    className="MuiList-root makeStyles-menuList-2 MuiList-padding"
    onKeyDown={[Function]}
    role="menu"
    tabIndex={-1}
  >
    <a
      className="makeStyles-link-4"
      href="/dashboard"
      onClick={[Function]}
      tabIndex={0}
    >
      <li
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-menuItem-3 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-6"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z"
            />
          </svg>
        </div>
        <span
          className="MuiTypography-root makeStyles-linkLabel-5"
        >
          Dashboard
        </span>
      </li>
    </a>
    <hr
      className="MuiDivider-root"
    />
    <a
      className="makeStyles-link-4"
      href="/reports"
      onClick={[Function]}
    >
      <li
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-menuItem-3 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-6"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 10H7v2h10v-2zm2-7h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11zm-5-5H7v2h7v-2z"
            />
          </svg>
        </div>
        <span
          className="MuiTypography-root makeStyles-linkLabel-5"
        >
          Relatórios
        </span>
      </li>
    </a>
    <hr
      className="MuiDivider-root"
    />
    <a
      className="makeStyles-link-4"
      href="/auditing"
      onClick={[Function]}
    >
      <li
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-menuItem-3 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-6"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14 10H2v2h12v-2zm0-4H2v2h12V6zM2 16h8v-2H2v2zm19.5-4.5L23 13l-6.99 7-4.51-4.5L13 14l3.01 3 5.49-5.5z"
            />
          </svg>
        </div>
        <span
          className="MuiTypography-root makeStyles-linkLabel-5"
        >
          Auditoria
        </span>
      </li>
    </a>
    <hr
      className="MuiDivider-root"
    />
    <a
      className="makeStyles-link-4"
      href="/users"
      onClick={[Function]}
    >
      <li
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-menuItem-3 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-6"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
            />
          </svg>
        </div>
        <span
          className="MuiTypography-root makeStyles-linkLabel-5"
        >
          Usuários
        </span>
      </li>
    </a>
    <hr
      className="MuiDivider-root"
    />
    <a
      className="makeStyles-link-4"
      href="/groups"
      onClick={[Function]}
    >
      <li
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-menuItem-3 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-6"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
            />
          </svg>
        </div>
        <span
          className="MuiTypography-root makeStyles-linkLabel-5"
        >
          Grupos
        </span>
      </li>
    </a>
    <hr
      className="MuiDivider-root"
    />
    <a
      className="makeStyles-link-4"
      href="/status"
      onClick={[Function]}
    >
      <li
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-menuItem-3 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-6"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm-2 14l-4-4 1.41-1.41L10 14.17l6.59-6.59L18 9l-8 8z"
            />
          </svg>
        </div>
        <span
          className="MuiTypography-root makeStyles-linkLabel-5"
        >
          Status
        </span>
      </li>
    </a>
    <hr
      className="MuiDivider-root"
    />
  </ul>
</div>
`;
